### PR TITLE
Inline `postJSON`. Improve docs.

### PR DESCRIPTION
**Pull request by rudolfbyker**

### CODE REVIEW

File: lib/solr.ts [hunk 1]
```diff
@@ -591,20 +584,46 @@ export class Client {
...
-      ipVersion: this.options.ipVersion,
-      // TODO this doesn't seem to be used
-      //request: this.request,
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json; charset=utf-8',
+        'content-length': Buffer.byteLength(json),
+        accept: 'application/json; charset=utf-8',
+      },
+      path: fullPath,
```
Comment (kibertoad) [inline lib/solr.ts:596]: we lose fullPath param here and replace it with path. Which one Request expects?

File: lib/solr.ts [hunk 2]
```diff
@@ -907,37 +926,63 @@ export class Client {
...
-      authorization: this.options.authorization,
-      agent: this.options.agent,
-      ipVersion: this.options.ipVersion,
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json; charset=utf-8',
+        'content-length': Buffer.byteLength(json),
+        accept: 'application/json; charset=utf-8',
+      },
+      path: fullPath,
```
Comment (kibertoad) [inline lib/solr.ts:960]: same, we replace fullPath with path

File: lib/solr.ts (hunk 1 above)
Comment (rudolfbyker) [reply lib/solr.ts:596]: We don't lose anything. It's the exact same thing as we had before: https://github.com/lbdremy/solr-node-client/pull/278/files/3418634b61cb16b6dca20dad0a8f9d469efa5503#diff-b0d5c84599fa1c42597f5bfad83e01676fdbe2cb87068ed1f57112293b54bd67L108

File: lib/solr.ts (hunk 2 above)
Comment (rudolfbyker) [reply lib/solr.ts:960]: Ditto. See https://github.com/lbdremy/solr-node-client/pull/278/files/3418634b61cb16b6dca20dad0a8f9d469efa5503#diff-b0d5c84599fa1c42597f5bfad83e01676fdbe2cb87068ed1f57112293b54bd67L108

File: lib/solr.ts (hunk 1 above)
Comment (rudolfbyker) [reply lib/solr.ts:596]: Also, fullPath is just the name of a variable. You could rename it to `path` or even `foo` if you wanted. See https://github.com/lbdremy/solr-node-client/pull/278/files/3418634b61cb16b6dca20dad0a8f9d469efa5503#diff-3b89c2e1c7e4e714474d6a2f17b55814a790482f4ef375d4f84d4895554c20d9L581

**Review by kibertoad — APPROVED** [review]

**Pull request merged**